cpp/modmesh/pilot/RWorld.cpp

```diff
@@ -28,34 +28,116 @@
-    m_renderer->setPrimitiveType(Qt3DRender::QGeometryRenderer::Lines);
-    addComponent(m_renderer);
-    addComponent(m_material);
+    size_t const npoint = world->nvertex();
+
+    if (npoint > 0)
+    {
+        {
+            // Build the Qt node (vertex) coordinate buffer.
+            auto * vertices = new Qt3DCore::QAttribute(m_geometry);
+
+            vertices->setName(Qt3DCore::QAttribute::defaultPositionAttributeName());
+            vertices->setAttributeType(Qt3DCore::QAttribute::VertexAttribute);
+            vertices->setVertexBaseType(Qt3DCore::QAttribute::Float);
+            vertices->setVertexSize(3);
+
+            auto * buf = new Qt3DCore::QBuffer(m_geometry);
+            {
+                QByteArray barray;
+                barray.resize(npoint * 3 * sizeof(float));
+                SimpleArray<float> sarr = makeSimpleArray<float>(barray, small_vector<size_t>{npoint, 3}, /*view*/ true);
+                for (size_t i = 0; i < world->nvertex(); ++i)
+                {
+                    Vector3dFp64 const & v = world->vertex(i);
+                    sarr(i, 0) = v[0];
+                    sarr(i, 1) = v[1];
+                    sarr(i, 2) = v[2];
+                }
+                buf->setData(barray);
+            }
+            vertices->setBuffer(buf);
+            vertices->setByteStride(3 * sizeof(float));
+            vertices->setCount(npoint);
+
+            m_geometry->addAttribute(vertices);
+        }
+
+        {
+            // Build the Qt node index buffer.
+            auto * indices = new Qt3DCore::QAttribute(m_geometry);
+
+            indices->setVertexBaseType(Qt3DCore::QAttribute::UnsignedInt);
+            indices->setAttributeType(Qt3DCore::QAttribute::IndexAttribute);
+
+            auto * buf = new Qt3DCore::QBuffer(m_geometry);
+            {
+                QByteArray barray;
+                barray.resize(npoint * sizeof(uint32_t));
+                SimpleArray<uint32_t> sarr = makeSimpleArray<uint32_t>(barray, small_vector<size_t>{npoint}, /*view*/ true);
+                for (size_t i = 0; i < world->nvertex(); ++i)
+                {
+                    sarr(i) = i;
+                }
+                buf->setData(barray);
+            }
+            indices->setBuffer(buf);
+            indices->setCount(npoint);
+
+            m_geometry->addAttribute(indices);
+        }
+
+        m_renderer->setGeometry(m_geometry);
+        m_renderer->setPrimitiveType(Qt3DRender::QGeometryRenderer::Points);
+
+        addComponent(m_renderer);
+
+        // Update material
+        {
+            auto effect = m_material->effect();
+            for (Qt3DRender::QTechnique * t : effect->techniques())
+            {
+                for (Qt3DRender::QRenderPass * rp : t->renderPasses())
+                {
+                    auto ps = new Qt3DRender::QPointSize(m_material);
+                    ps->setSizeMode(Qt3DRender::QPointSize::SizeMode::Fixed);
+                    ps->setValue(4.0f);
```

Comment:
This is critical to show the points.  Otherwise the points do not show, maybe having 0 size?